roles/network_plugin/calico/tasks/install.yml

```diff
@@ -571,7 +571,7 @@
     - "{{ peers|selectattr('scope','undefined')|list|default([]) | union(peers|selectattr('scope','defined')|selectattr('scope','equalto', 'node')|list|default([])) }}"
   when:
     - peer_with_router|default(false)
-    - inventory_hostname == groups['kube_control_plane'][0]
```

Comment:
I updated this PR with a `delegate_to` to ensure that all `calicoctl.sh` calls are done on the first node. This is a different behavior than the one that was in place before #8833 , but I has the same effects so it's fine by me.

I'd like more details about the part of your answer that states that `calicoctl.sh` is not configured everywhere: in my understanding, this script is deployed (and functional) on all cluster nodes. This is why the task that this PR modifies was working up to the recent changes. 
Is this not supposed to be the case anymore? Is `calicoctl.sh` planned to only be available on control plane nodes?

Reply:
You are correct that calicoctl.sh is deployed on all nodes, I was under a wrong impression that we only deployed it on control plane nodes. Since we render and apply manifests only on the first control plane node, it makes sense to have the same approach for calico commands even though they could be ran from any node. In the end I'm impartial to the implementation but open to other perspectives.

/cc @floryut @oomichi WDYT ?